### PR TITLE
351 create sf jobs from tc

### LINKED ISSUE #351: Create SF jobs from TC
TC staff create jobs directly on SF at the moment. This doesn't work for recruiter access, who will not have access to SF.

We still need SF jobs to be created but this should all be done from the TC.

### PR DESCRIPTION
**Pull request by camerojo**

### CODE REVIEW

File: server/src/main/java/org/tctalent/server/model/db/AbstractSalesforceObject.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+/**
+ * Base class of entities with corresponding objects on Salesforce.
+ *
+ * @author John Cameron
+ */
+@Getter
+@Setter
+@MappedSuperclass
+public class AbstractSalesforceObject extends AbstractAuditableDomainObject<Long> {
```
Comment (sadatmalik) [inline server/src/main/java/org/tctalent/server/model/db/AbstractSalesforceObject.java:31]: `public abstract class AbstractSalesforceObject extends AbstractAuditableDomainObject<Long> {`

Comment (sadatmalik) [inline server/src/main/java/org/tctalent/server/model/db/AbstractSalesforceObject.java:31]:
```suggestion
public abstract class AbstractSalesforceObject extends AbstractAuditableDomainObject<Long> {
```

File: server/src/main/java/org/tctalent/server/model/db/AbstractOpportunity.java [hunk 2]
```diff
@@ -32,7 +32,7 @@
 @Getter
 @Setter
 @MappedSuperclass
-public class AbstractOpportunity extends AbstractAuditableDomainObject<Long> {
+public class AbstractOpportunity extends AbstractSalesforceObject {
```
Comment (sadatmalik) [inline server/src/main/java/org/tctalent/server/model/db/AbstractOpportunity.java:35]:
```suggestion
public abstract class AbstractOpportunity extends AbstractSalesforceObject {
```

**Review by sadatmalik — DISMISSED** [review]
Approved

File: server/src/main/java/org/tctalent/server/model/db/AbstractSalesforceObject.java (hunk 1 above)
Comment (camerojo) [reply server/src/main/java/org/tctalent/server/model/db/AbstractSalesforceObject.java:31]: I initially had this as abstract - but I recall it lead to JPA problems

**Pull request merged**